sdk/src/main/java/com/hedera/hashgraph/sdk/Hbar.java

```diff
@@ -64,14 +64,39 @@ public Hbar(BigDecimal amount) {
      */
     public static final Hbar MIN = Hbar.from(-50_000_000_000L);
 
+    private static int findNumber(CharSequence text) {
+        for(int i = 0; i < text.length(); i++) {
```

Comment:
Should just use `String.split()`

Reply:
Use https://regex101.com/r/6Q5EOm/1

Reply:
How about `.split(" ")`?

Let the `BigDecimal` constructor and `getUnit()` do the work of deciding if they're correctly formatted.

Reply:
Reason for regex is to validate the input makes sense before we try to do anything with it. What if someone submits ` 1 ℏ` This would split on the first space and cause `BigDecimal` to return `0` (I think) and the `getUnit()` method would error even though the unit wasn't the issue, it was the misformatted input. You _can_ use the regex to validate and then use the `.split(" ")` instead of using the regex groups if you want, but to re-iterate it's mainly for validating input.